### PR TITLE
[CARBONDATA-2317] Concurrent datamap with same name and schema creation throws exception

### PR DESCRIPTION
**Pull request by rahulforallp**
- [x] Any interfaces changed? No
 
 - [x] Any backward compatibility impacted? No
 
 - [x] Document update required? No

 - [x] Testing done
        NA. All UT and sdv success report only required.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4853/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3631/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4322/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3648/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4870/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4339/

**Comment by manishgupta88** [comment]
Add a test case for this scenario. We have multiple test cases to test concurrency scenarios, you can take the reference from there

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonCreateTableCommand.scala [hunk 1]
```diff
@@ -121,21 +121,24 @@ case class CarbonCreateTableCommand(
...
-               |  dbName "$dbName",
-               |  tablePath "$tablePath",
-               |  path "$tablePath",
-               |  isExternal "$isExternal",
-               |  isUnManaged "$isUnmanaged",
-               |  isVisible "$isVisible"
-               |  $carbonSchemaString)
-               |  $partitionString
+          // synchronized to prevent concurrently creation of table with same name
+          CarbonCreateTableCommand.synchronized {
```
Comment (manishgupta88) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonCreateTableCommand.scala]: CarbonCreateTableCommand is a case class and each thread will

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3686/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4909/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4375/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**